internal/platform/file.go

```diff
@@ -3,8 +3,83 @@ package platform
 import (
 	"io"
 	"io/fs"
+	"syscall"
 )
 
+// File is a writeable fs.File bridge backed by syscall functions needed for ABI
+// including WASI and runtime.GOOS=js.
+//
+// Implementations should embed UnimplementedFile for forward compatability. Any
+// unsupported method or parameter should return syscall.ENOSYS.
+//
+// # Errors
+//
+// All methods that can return an error return a syscall.Errno, which is zero
+// on success.
+//
+// Restricting to syscall.Errno matches current WebAssembly host functions,
+// which are constrained to well-known error codes. For example, `GOOS=js` maps
+// hard coded values and panics otherwise. More commonly, WASI maps syscall
+// errors to u32 numeric values.
+//
+// # Notes
+//
+// A writable filesystem abstraction is not yet implemented as of Go 1.20. See
+// https://github.com/golang/go/issues/45757
+type File interface {
+	// Stat is similar to syscall.Fstat.
+	//
+	// # Errors
+	//
+	// The following errors are expected:
+	//   - syscall.EBADF if the file or directory was closed.
+	//
+	// # Notes
+	//
+	//   - An fs.FileInfo backed implementation sets atim, mtim and ctim to the
+	//     same value.
+	//   - Windows allows you to stat a closed directory.
+	Stat() (Stat_t, syscall.Errno)
```

Comment:
so basically this stubs in by only defining stat and close. Main point is to carve in a way to migrate (to anything). then if it works, multiple people can work on whatever the direction is easier.